### DISCUSSION TITLE
Is there support for transactions?

### DISCUSSION THREAD

**Discussion by felinto-dev** [RobertCraigie/prisma-client-py#334] [Q&A] [answered]
Hey there! I look the documentation but I didn't find any related. Maybe is there a helper utility I can import from the Prisma client?

I would to add bulk inserts to the database. All elements should be inserted OR all fail. A common use case for transactions.

**Comment by RobertCraigie** [comment, marked as answer]
Sounds like [query batching](https://prisma-client-py.readthedocs.io/en/stable/reference/batching/) or [create_many](https://prisma-client-py.readthedocs.io/en/stable/reference/operations/#many-records) would be the solution for you.

The limitation with query batching is that the records are sent all at once which means you can't create one record then create another record that relies on the first one for example. In this case you would need interactive transactions which are not supported yet but are being worked on #53.

**Reply by felinto-dev** [reply]
works like magic! thanks!
I used query batching.
